cmd/main.go

```diff
@@ -233,6 +235,7 @@ func startServiceController(mgr manager.Manager, nsxClient *nsx.Client) {
 	}
 
 	if restoreMode {
+		subnetSetReconcile.EnableRestoreMode()
```

Comment:
In the other reconciler, we set `restoreMode = true` in the `RestoreReconcile` itself. Could you help explain why the subnetset is different?

Reply:
Sure, it's because in SubnetSet GC, there is a step will update status of SubnetSet for scaling down case. It will delete the SubnetInfo in subnetset.status.subnets if the corresponding NSX Subnet is not in store, while in restore mode we would like to keep this SubnetInfo as source of truth to restore the NSX Subnet.

https://github.com/vmware-tanzu/nsx-operator/blob/1816f9fc78583356aa38c54fb94b56f1a80d3fec/pkg/controllers/subnetset/subnetset_controller.go#L390-L393

Thus we need a flag in GC to skip the status update when a SubnetInfo without SubnetPort exists. As GC is called before RestoreReconcile, I set the restoreMode to true in main unlike other reconciler.

Reply:
Considering NSX Operator will restart after the restore process ends, I think it might be better to remove the `restoreMode=false` in other reconcilers as there is no use case. Thanks for the reminder!